harper-core/src/linting/that_which.rs

```diff
@@ -0,0 +1,68 @@
+use itertools::Itertools;
+
+use crate::{
+    patterns::{Pattern, SequencePattern, WordPatternGroup},
+    Token, TokenStringExt,
+};
+
+use super::{Lint, LintKind, PatternLinter, Suggestion};
+
+pub struct ThatWhich {
+    pattern: Box<dyn Pattern>,
+}
+
+impl Default for ThatWhich {
+    fn default() -> Self {
+        let mut pattern = WordPatternGroup::default();
+
+        pattern.add(
+            "that",
+            Box::new(
```

Comment:
Used twice -> refactor into own variable.

Reply:
You can solve this without using `Clone`. A smart-pointer for example.